pkg/storer/internal/reserve/reserve.go

```diff
@@ -145,95 +164,77 @@ func (r *Reserve) Put(ctx context.Context, chunk swarm.Chunk) error {
 			if err != nil {
 				return err
 			}
-			prev := binary.BigEndian.Uint64(sameAddressOldStampIndex.StampTimestamp)
-			curr := binary.BigEndian.Uint64(chunk.Stamp().Timestamp())
-			if prev >= curr {
-				return fmt.Errorf("overwrite same chunk. prev %d cur %d batch %s: %w", prev, curr, hex.EncodeToString(chunk.Stamp().BatchID()), storage.ErrOverwriteNewerChunk)
-			}
 
-			// index collision with another chunk
-			if loadedStampIndex {
-				prev := binary.BigEndian.Uint64(oldStampIndex.StampTimestamp)
+			// same index
+			if bytes.Equal(sameAddressOldStamp.Index(), chunk.Stamp().Index()) {
```

Comment:
Even in this same address condition, you also need to check if the new stamp is overwriting a different unrelated stamp index.

Reply:
https://github.com/ethersphere/bee/blob/master/pkg/storer/internal/reserve/reserve.go#L155C7-L155C23

Reply:
that should be handled below, shouldn't it? (it does not return if there is no index collision)
anyway the tests are passing it should be fine.